tests/chainer_tests/test_reporter.py

```diff
@@ -216,5 +217,208 @@ def test_float(self):
         testing.assert_allclose(mean, 2.)
         testing.assert_allclose(std, numpy.sqrt(2. / 3.))
 
+    def test_serialize(self):
+        self.summary.add(1.)
+        self.summary.add(2.)
+
+        summary = chainer.reporter.Summary()
+        testing.save_and_load_npz(self.summary, summary)
+        summary.add(3.)
+
+        mean = summary.compute_mean()
+        testing.assert_allclose(mean, 2.)
+
+        mean, std = summary.make_statistics()
+        testing.assert_allclose(mean, 2.)
+        testing.assert_allclose(std, numpy.sqrt(2. / 3.))
+
+    @attr.gpu
+    def test_serialize_cupy(self):
+        xp = cuda.cupy
+        self.summary.add(xp.array(1, 'f'))
+        self.summary.add(xp.array(2, 'f'))
+
+        summary = chainer.reporter.Summary()
+        testing.save_and_load_npz(self.summary, summary)
+        summary.add(xp.array(3, 'f'))
+
+        mean = summary.compute_mean()
+        testing.assert_allclose(mean, 2.)
+
+        mean, std = summary.make_statistics()
+        testing.assert_allclose(mean, 2.)
+        testing.assert_allclose(std, numpy.sqrt(2. / 3.))
+
+    def test_serialize_backward_compat(self):
+        with tempfile.NamedTemporaryFile(delete=False) as f:
```

Comment:
Without `delete=False`, the temporary file is deleted after `numpy.savez` because it closes `f`.

Reply:
I think `numpy.savez` does not close the file.

```
>>> import numpy as np
>>> f = open('test.dat', 'wb')
>>> np.savez(f, np.array(10))
>>> np.savez(f, np.array(10))
```

Reply:
https://ci.appveyor.com/project/pfnet/chainer/build/8001/job/8nxy5iuv60n8wla8
AppVeyor fails without `delete=False`

Reply:
https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).

Reply:
Thanks. I wasn't aware that open temporary files cannot be reopened on Windows.
Could you please revert `delete=True` commit?  Sorry to trouble you.

Reply:
Sure